### PR TITLE
[RICV-V]Scope linear cross entropy ULP tolerance to RISC-V (#42)

### PR DESCRIPTION
**Pull request by WuXintong123**
This PR updates test/test_nn.py to make linear_cross_entropy CPU fp32 input-gradient ULP tolerance RISC-V aware. It uses IS_RISCV64 to set the CPU threshold to 512 on RISC-V (from the previous fixed 384), while keeping 384 on other CPUs, preventing unnecessary false test failures on RISC-V.

### CODE REVIEW

**Comment by albanD** [comment]
Thank you very much for sending a PR.
To avoid you spending time preparing a PR for a feature we're not sure we want to add, our usual process is to discuss on the issue before doing the PR.
In particular, make sure that the issue is marked "actionable" before sending a PR!
See https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions for details!
